### ISSUE TITLE
Complete Azure storage REST API

### ISSUE THREAD

**Issue by MindFlavor** [MindFlavor/AzureSDKForRust#73]
This will be done after [https://github.com/MindFlavor/AzureSDKForRust/issues/61](https://github.com/MindFlavor/AzureSDKForRust/issues/61):

- [ ] Undelete blob
- [ ] Get blob properties
- [ ] Set blob properties
- [ ] Get blob metadata
- [ ] Set blob metadata
- [ ] Snapshot blob
- [ ] Copy blob
- [ ] Abort blob
- [ ] Set blob tier
- [ ] Undelete blob
- [ ] Get page ranges
- [ ] Incremental copy blob (page blob)
- [ ] Append block (append blob)

**Comment by mpbraendli** [comment]
Hello, I've been using your blob functionality successfully, although my usage was only storing blobs. I now need to use [Copy Blob From URL](https://docs.microsoft.com/en-us/rest/api/storageservices/copy-blob-from-url)

How can I help you implement this API into the SDK?

I guess the first step is to create a new request in "AzureSDKForRust/azure_sdk_storage_blob/src/blob/requests". Is it fine for you if I start looking into it?

**Comment by MindFlavor** [comment]
Yes of course, any help is appreciated! I might be able to take a break from the Cosmos create now (I've been focusing on than lately) and tackle this but I cannot guarantee on the timeline!

As for your question, in order to extend the crate you need:

1. Create the corresponding request (in your case in [`blob/request`](https://github.com/MindFlavor/AzureSDKForRust/tree/master/azure_sdk_storage_blob/src/blob/requests)). You can use the [bpb](https://github.com/MindFlavor/bpb) tool if you want to start from the json and save yourself a lot of boilerplate code.
2. Create the corresponding response (in your case in [`blob/responses`](https://github.com/MindFlavor/AzureSDKForRust/tree/master/azure_sdk_storage_blob/src/blob/responses)). Check other responses for a template.
3. Have the `finalize` function return a `Result<your_response_from_point_2, AzureError>`. Make sure to pass the mandatory and optional headers and construct the URI properly (that's the hard part).
4. Add the signature to the correct trait (in your case `Blob` trait in [lib.rs](https://github.com/MindFlavor/AzureSDKForRust/blob/master/azure_sdk_storage_blob/src/lib.rs)).
5. Implement the trait function in the `Client` just returning the generator created at point 1.

After that you should be able to use your function in your code!

**Comment by MindFlavor** [comment]
This PR should give you the function you need: https://github.com/MindFlavor/AzureSDKForRust/pull/258. It's almost done, I just need to fix the crate version and I will merge it.

**Comment by MindFlavor** [comment]
Implemented in [https://github.com/MindFlavor/AzureSDKForRust/releases/tag/blob_0.43.1](https://github.com/MindFlavor/AzureSDKForRust/releases/tag/blob_0.43.1)!

**Comment by mpbraendli** [comment]
Really cool, thank you so much!
